### PR TITLE
Update links and references in the contributing guidelines and license.

### PR DESCRIPTION
**Pull request by hodgestar**
### Summary

I've updated the repository links in the contributing guidelines and the copyright years in the license file.

The contributing guidelines referred to a now non-existent template for proposing specification changes, so I added a new template.

Update:

I found a few other places that referenced the old repository so I updated those. Notably:

- I changed the repo URL in `setup.cfg`
- I removed what appears to be an outdated caveat in the OpenPulse specification that referred to a closed issue.

### Details and comments

I think everything is covered above in the summary.

There is a question about what the copyright dates should be. I opted to cover the date range from when the license file was added (which seems a reasonable year for first publication) until now (since each commit essentially publishes the changes).

### CODE REVIEW

File: source/language/openpulse.rst [hunk 1]
```diff
@@ -238,7 +238,7 @@ Waveforms
 
 Waveforms are of type ``waveform`` and can either be:
 
-- An array of complex samples (note this syntax is still under [active development](https://github.com/Qiskit/openqasm/pull/301) and is subject to change.) which define the points for the waveform envelope
+- An array of complex samples which define the points for the waveform envelope
```
Comment (jakelishman) [inline source/language/openpulse.rst:241]: I don't feel confident enough on the openpulse development to approve this change, but I suspect it's fine.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -56,19 +56,19 @@ Please follow the next rules for the commit messages:
...
 ## Spec proposals
 
-:bulb: All new ideas go through the next stages to become a new feature of the language.
+:bulb: All new ideas go through the following stages to become a new feature of the language:
 
-- Draft: Backlog items with different levels of abstraction. Anybody can add one issue in the [main repo](https://github.com/Qiskit/openqasm). Please label it as `draft`.
+- Draft: Backlog items with different levels of abstraction. Anybody can add an issue in the [main repo](https://github.com/openqasm/openqasm). Please label it as `draft`.
 - Proposal: An idea with the correct form:
-  - Add one issue in the [main repo](https://github.com/Qiskit/openqasm) labeled as `proposal` using [this template](templates/proposal.md).
+  - Add an issue in the [main repo](https://github.com/openqasm/openqasm) labeled as `proposal` using the spec proposal template.
```
Comment (jakelishman) [inline CONTRIBUTING.md:69]:
```suggestion
  - Add an issue in the [main repo](https://github.com/openqasm/openqasm) labeled as `proposal` using [the spec proposal template](https://github.com/openqasm/openqasm/issues/new?template=spec_proposal.yaml)
```

I _think_ but am not sure that this link should directly open the "new issue" dialogue with the new form.

**Review by jakelishman — COMMENTED** [review]
All the link fixing looks good to me, though I don't want to unilaterally approve a new "propose an addition to the spec" template.

File: source/language/openpulse.rst (hunk 1 above)
Comment (hodgestar) [reply source/language/openpulse.rst:241]: The pull request referenced is closed, so the text should change somehow, but happy to change it another way if someone has a different suggestion.

File: CONTRIBUTING.md (hunk 2 above)
Comment (hodgestar) [reply CONTRIBUTING.md]: Thanks. I tested the link on my private GitHub trying things out repo and it works nicely.

**Comment by hodgestar** [comment]
Thanks @jakelishman. Let's wait for a couple more eyes to have a look.

File: source/language/openpulse.rst (hunk 1 above)
Comment (braised-babbage) [reply source/language/openpulse.rst:241]: For what it's worth, in our implementation of OpenPulse we've moved away from this style of waveform declaration. I wrote up a few more details here https://github.com/openqasm/openqasm/pull/414#issuecomment-1839633685.

**Review by jakelishman — APPROVED** [review]
I'm happy to reapply my tick, and Erik, if you're happy that Simon's change to the OpenPulse comment is fine, I think it's fine to merge (I'll leave you to press the button, in case you disagree).  The technical detail on potential changes to the OpenPulse spec I guess would be in your separate issue/PR.

**Comment by braised-babbage** [comment]
Looks good to me. Some of these changes are long overdue -- thanks for the PR!

**Comment by hodgestar** [comment]
Thanks @braised-babbage and @jakelishman.

**Pull request merged**